src/sailfishusermanager.cpp

```diff
@@ -361,7 +362,18 @@ uint SailfishUserManager::addSailfishUser(const QString &user, const QString &na
 
     // Execute user creation scripts
     QDir creation(USER_CREATE_SCRIPT_DIR);
-    for (const QString &entry : creation.entryList(QDir::Files | QDir::Executable)) {
+    creation.setSorting(QDir::NoSort);
+    creation.setFilter(QDir::Files | QDir::Executable);
+
+    auto entryList = creation.entryList();
+
+    QCollator collator;
+    collator.setNumericMode(true);
+    collator.setCaseSensitivity( Qt::CaseInsensitive );
+
+    std::sort(entryList.begin(), entryList.end(), collator);
```

Comment:
This fine-tuned sorting makes adding and removing users behave differently. And this filter+sort variant seems to have problems that remove avoids e.g. using "*.sh" pattern ensures random files that were accidentally installed are not executed (say, editor backup files, either due to on-device ad-hoc testing or hiccups in rpm packaging).

I'd suggest:

- instead of inlining execute loop here, implement and use createUserFiles()  analogous to already existing removeUserFiles()
- would not hurt if both would use common code for the execute-scripts-in-dir bit
- but in any case: if more fine tuned sorting is needed, it should be used both for create and remove scripts